tinydir.h

```diff
@@ -658,19 +658,29 @@ int tinydir_file_open(tinydir_file *file, const _tinydir_char_t *path)
 	/* Get the parent path */
 #if (defined _MSC_VER || defined __MINGW32__)
 #if ((defined _MSC_VER) && (_MSC_VER >= 1400))
-		_tsplitpath_s(
+		errno = _tsplitpath_s(
 			path,
 			drive_buf, _TINYDIR_DRIVE_MAX,
 			dir_name_buf, _TINYDIR_FILENAME_MAX,
 			file_name_buf, _TINYDIR_FILENAME_MAX,
 			ext_buf, _TINYDIR_FILENAME_MAX);
+		if (errno)
+		{
+			return -1;
+		}
 #else
 		_tsplitpath(
 			path,
 			drive_buf,
 			dir_name_buf,
 			file_name_buf,
 			ext_buf);
+
+		if (errno)
+		{
+			errno = EINVAL;
```

Comment:
this line isn't necessary anymore, `_tsplitpath` only uses `EINVAL` according to the docs.
This also means this `if (errno)` block doesn't need to be moved inside the `#ifdef` block.

Reply:
The problem described by issue #58 is in `if (errno)`. When I check return value of `_tsplitpath_s` it was zero, but errno is set already.
According to docs `_tsplitpath`:
> If path is NULL, the invalid parameter handler is invoked, as described in Parameter Validation. If execution is allowed to continue, errno is set to EINVAL and the function returns EINVAL.

`_tsplitpath` sets `errno` and looks like `_tsplitpath` is not `(void)`
and from docs `_tsplitpath_s`:
> Return Value
Zero if successful; an error code on failure.

May be we need to check retval? not `errno`?